src/include/distributed/multi_physical_planner.h

```diff
@@ -221,6 +221,11 @@ typedef struct MultiPlan
 	Query *masterQuery;
 	bool routerExecutable;
 
+	/* INSERT ... SELECT via coordinator only */
+	Query *insertSelectQuery;
```

Comment:
Minor discussion: When I see `insertSelectQuery`, I feel that it covers both the distributed and the coordinator one. So, thinking to suggest renaming the field to `coordinatorInsertSelectQuery` (or so), but that's a very long name for a variable too. So, just sharing my thoughts :)